### PR TITLE
listmonk: modify the 1.20 Go support patch

### PR DESCRIPTION
**Pull request by RaitoBezarius**
## Description of changes

Previously, it was a PR that was merged, meaning this will break. Added a comment to explain why this patch.

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by Janik-Haag** [comment]
Result of `nixpkgs-review pr 251819` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>listmonk</li>
  </ul>
</details>

**Review by Janik-Haag — APPROVED** [review]
LGTM

File: pkgs/servers/mail/listmonk/default.nix [hunk 1]
```diff
@@ -12,8 +12,9 @@ buildGoModule rec {
   };
 
   patches = [
+    # Ensure that listmonk supports Go 1.20
     (fetchpatch {
-      url = "https://github.com/knadh/listmonk/pull/1479.patch";
+      url = "https://github.com/knadh/listmonk/commit/25513b81044803b104ada63c0be57a913960484e.patch";
       hash = "sha256-SYACM8r+NgeSWn9VJV4+wkm+6s/MhNGwn5zyc2tw7FU=";
     })
```
Comment (Janik-Haag) [inline pkgs/servers/mail/listmonk/default.nix:19]: I think adding the pr number or a link to the pr might be a good idea so it's easier for people to figure out that they can omit the patch when they update the package.

Comment (RaitoBezarius) [reply pkgs/servers/mail/listmonk/default.nix:19]: It has been merged though, so the question is rather: is this commit part of an release, no?

**Pull request merged**